fpmsyncd/routesync.cpp

```diff
@@ -547,6 +810,606 @@ void RouteSync::onEvpnRouteMsg(struct nlmsghdr *h, int len)
+        {
+            SWSS_LOG_ERROR("Fail to get the VRF name (ifindex %u)", vrf_index);
+            return;
+        }
+        /*
+         * Now vrf device name is required to start with VRF_PREFIX
+         */
+        if (memcmp(routeTableKey, VRF_PREFIX, strlen(VRF_PREFIX)))
+        {
+            SWSS_LOG_ERROR("Invalid VRF name %s (ifindex %u)", routeTableKey,
+                           vrf_index);
+            return;
+        }
+        routeTableKey[strlen(routeTableKey)] = ':';
+    }
+
+    if ((rtm->rtm_family == AF_INET && dst_len == IPV4_MAX_BITLEN) ||
+        (rtm->rtm_family == AF_INET6 && dst_len == IPV6_MAX_BITLEN))
+    {
+        snprintf(routeTableKey + strlen(routeTableKey),
+                 sizeof(routeTableKey) - strlen(routeTableKey), "%s",
+                 destipprefix);
+    }
+    else
+    {
+        snprintf(routeTableKey + strlen(routeTableKey),
+                 sizeof(routeTableKey) - strlen(routeTableKey), "%s/%u",
+                 destipprefix, dst_len);
+    }
+
+    SWSS_LOG_INFO("Received route message dest ip prefix: %s Op:%s",
+                  destipprefix, nlmsg_type == RTM_NEWROUTE ? "add" : "del");
+
+    /*
+     * In the current FPM implementation, a RTM_DELROUTE message need not be
+     * accompanied by any nexthops, therefore we expect all RTM_DELROUTE
+     * messages to be processed by onMsg() using the rtnl api.
+     */
+    if (nlmsg_type == RTM_DELROUTE)
+    {
+        SWSS_LOG_ERROR("RTM_DELROUTE message-type not expected (%s)",
+                       destipprefix);
+        return;
+    }
+    else if (nlmsg_type != RTM_NEWROUTE)
+    {
+        SWSS_LOG_ERROR("Unknown message-type: %d for %s", nlmsg_type,
+                       destipprefix);
+        return;
+    }
+
+    switch (rtm->rtm_type)
+    {
+        case RTN_BLACKHOLE:
+        case RTN_UNREACHABLE:
+        case RTN_PROHIBIT:
+            SWSS_LOG_ERROR(
+                "RTN_BLACKHOLE route not expected (%s)", destipprefix);
+            return;
+        case RTN_UNICAST:
+            break;
+
+        case RTN_MULTICAST:
+        case RTN_BROADCAST:
+        case RTN_LOCAL:
+            SWSS_LOG_NOTICE(
+                "BUM routes aren't supported yet (%s)", destipprefix);
+            return;
+
+        default:
+            return;
+    }
+
+    /* Get nexthop lists */
+    string vpn_sid_str;
+    string src_addr_str;
+    bool ret;
+
+    ret = getSrv6SteerRouteNextHop(h, len, tb, vpn_sid_str, src_addr_str);
+    if (ret == false)
+    {
+        SWSS_LOG_NOTICE(
+            "SRv6 Route issue with RouteTable msg: %s vpn_sid:%s src_addr:%s",
+            destipprefix, vpn_sid_str.c_str(), src_addr_str.c_str());
+        return;
+    }
+
+    if (vpn_sid_str.empty())
+    {
+        SWSS_LOG_NOTICE("SRv6 IP Prefix: %s vpn_sid is empty", destipprefix);
+        return;
+    }
+
+    vector<FieldValueTuple> fvVectorRoute;
+    if (!vpn_sid_str.empty())
```

Comment:
this is not required since this check is already performed few lines above

Reply:
I've removed it, thanks!